# AlphaFold Usage Example

This notebook demonstrates how to use AlphaFold through the `pyeed` package interface.

## Setup and Imports

In [1]:
import os

import logging
from pyeed.tools.alphafold2 import AlphaFoldRunner

In [2]:
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
LOGGER = logging.getLogger(__name__)

Configures logging to display timestamped INFO-level messages.

## Input Preparation

In [3]:
id = 'AAP20891.1'
sequence = 'MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFKVLLCGAVLSRVDAGQEQLGRRIHYSQNDLVEYSPVTEKHLTDGMTVRELCSAAITMSDNTAANLLLTTIGGPKELTAFLHNMGDHVTRLDRWEPELNEAIPNDERDTTMPAAMATTLRKLLTGELLTLASRQQLIDWMEADKVAGPLLRSALPAGWFIADKSGAGERGSRGIIAALGPDGKPSRIVVIYTTGSQATMDERNRQIAEIGASLIKHW'

In [4]:
len(sequence)

286

Verifies the length of the input sequence (286 amino acids).

## AlphaFold Configuration

In [5]:
data_dir = '/media/database/alphafold' # the path to the downloaded alphafold data
output_dir = os.path.join(os.path.dirname(os.getcwd()), "resources", "alphafold", "output")

Sets up the paths for AlphaFold database and output directory.


In [6]:
alphafold_runner = AlphaFoldRunner(data_dir, output_dir)

2025-02-09 18:21:49,187 - INFO - GPU detected. AlphaFold will run on GPU.


Initializes the AlphaFold runner with the specified directories. The system detected a GPU for computation.

## Execution


In [7]:
alphafold_runner.run_alphafold(sequence=sequence, sequence_id=id)

2025-02-09 18:21:49,193 - INFO - Created FASTA file at: /home/nab/Niklas/pyeed/docs/resources/alphafold/output/AAP20891_1.fasta
2025-02-09 18:21:49,194 - INFO - Running AlphaFold with command: source ~/anaconda3/etc/profile.d/conda.sh && conda activate alphafold_env && python /home/nab/Niklas/pyeed/src/pyeed/tools/resources/alphafold/docker_run.py --fasta_paths=/home/nab/Niklas/pyeed/docs/resources/alphafold/output/AAP20891_1.fasta --max_template_date=2022-01-01 --data_dir=/media/database/alphafold --output_dir=/home/nab/Niklas/pyeed/docs/resources/alphafold/output
2025-02-09 18:59:25,222 - ERROR - AlphaFold stderr:
I0209 18:21:49.823744 126234918065984 docker_run.py:143] Mounting /home/nab/Niklas/pyeed/docs/resources/alphafold/output -> /mnt/fasta_path_0
I0209 18:21:49.823894 126234918065984 docker_run.py:143] Mounting /media/database/alphafold/uniref90 -> /mnt/uniref90_database_path
I0209 18:21:49.823978 126234918065984 docker_run.py:143] Mounting /media/database/alphafold/mgnify -> 

{}

Attempts to run AlphaFold prediction on the input sequence. The output indicates that a FASTA file was created, but the execution encountered an error when trying to locate the output structure file.

## Note
The execution appears to have encountered an error when trying to find the output PDB file (`AAP20891_1_relaxed.pdb`). This might indicate an issue with the AlphaFold execution or output file generation.